doc/HOST_PHASES.md

```diff
@@ -0,0 +1,99 @@
+# Host Phases
+
+The `ElementalHost.status.phase` describes the current phase the host is going through.  
+Host phases can be used to quickly determine what any `ElementalHost` is doing, for example if it is still installing, resetting, or just running normally.  
+
+All `ElementalHosts` are following the same phases, however the implementation for each phase heavily depends on the [OSPlugin](../cmd/agent/README.md#plugins) in use.  
+Please refer to the plugin documentation to learn about specific details. For example the [Elemental plugin](../cmd/agent/PLUGIN_ELEMENTAL.md).  
+
+## Phases
+
+Phases are normally executed in the documented order, but this might not always be the case.  
+For example a host can be reset just after installation, or some hosts will never use the `Bootstrapping` phase, if not associated to any CAPI Machine.  
+
+Most importantly, during normal operation the `Triggering Reset` phase is prioritized, so that in case of any errors, the user has the possibility to execute a host reset, to fix issues, for example to recover from a failed bootstrap.  
+
+### Registering
+
+The `Registering` phase is the first ever phase for any host.  
+During this phase the `elemental-agent` picks a new hostname (according to the `ElementalRegistration` configuratio), creates a new key pair for identification, and attempts to create a new `ElementalHost` on the management cluster using the [Elemental API](./ELEMENTAL_API_SETUP.md).  
+If an `ElementalHost` with the same name and public key already exists, the `elemental-agent` will consider the registration already done. This allows to attempt registration multiple times, **when not using random hostnames**.  
+
+Normally the `Registering` phase is very short living and transitory.  
+Upon successful registration, the `elemental-agent` will automatically execute the `Finalizing Registration` phase.  
+
+This phase is ran using the `elemental-agent --register` argument.  
+
+### Finalizing Registration
+
+The `Finalizing Registration` immediately follows the `Registration` phase when `elemental-agent --register` is used.  
+
+During this phase the `elemental-agent` will take the following steps:  
+
+1. Install the registered hostname. (OSPlugin dependent)
+
+1. Fetch the remote `ElementalRegistration` and use it to install a new agent config file. (OSPlugin dependent)  
+**Note:** During this phase the current agent config path is used to determine the install location. This is important if you are using a custom path (ex. `elemental-agent --config /my/custom/config.yaml`) and you will need to change it later. Migration of this file is going to be needed and depending on how the `OSPlugin` installs files (for ex. in an immutable system), the migration strategy may differ.  
+
+1. Installs the generated private key used for the host registration. (OSPlugin dependent)
```

Comment:
Since it is a step, I would drop the third person (as the other points)
```suggestion
1. Install the generated private key used for the host registration. (OSPlugin dependent)
```